UPGRADE_NOTES.md

```diff
@@ -1,3 +1,66 @@
+## Upgrading from 0.23 to 0.24
+
+For a full diff of necessary changes, please see [LuckyDiff](https://luckydiff.com?from=0.23.0&to=0.24.0).
+
+- Upgrade Lucky CLI (homebrew)
+
+```
+brew update
+brew upgrade lucky
+```
+
+- Upgrade Lucky CLI (Linux)
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Crystal should be `0.35.1`
+  - Lucky should be `~> 0.24.0`
+
+- Run `shards update`
+
+### General updates
+
+- Rename: all instances of the `m` macro to `mount`. e.g. `m Shared::Footer, year: 2020` -> `mount Shared::Footer, year: 2020`.
+- Update: `config/database.cr` with new `Avram::Credentials`.
+```crystal
+AppDatabase.configure do |settings|
+  if Lucky::Env.production?
+    settings.credentials = Avram::Credentials.parse(ENV["DB_URL"])
+  else
+    settings.credentials = Avram::Credentials.parse?(ENV["DB_URL"]?) || Avram::Credentials.new(
+      database: database_name,
+      hostname: ENV["DB_HOST"]? || "localhost",
+      port: ENV["DB_PORT"]?.try(&.to_i) || 5432,
```

Comment:
I think it'd be good to highlight this `try.(&.to_i)` specifically since that is likely to bite people